src/server/routes/welcome.ts

```diff
@@ -535,4 +583,71 @@ const OktaResendEmail = async (req: Request, res: ResponseWithRequestState) => {
 	}
 };
 
+const updateNewslettersAndConstents = async (
+	registrationConsents: RegistrationConsents,
+	res: ResponseWithRequestState,
+	loggingContext: string,
+) => {
+	const runningInCypress = process.env.RUNNING_IN_CYPRESS === 'true';
+	const state = res.locals;
+
+	if (!state.oauthState) {
+		return;
+	}
+
+	if (registrationConsents.consents?.length) {
+		try {
+			await updateConsents({
+				accessToken: state.oauthState.accessToken.toString(),
+				payload: registrationConsents.consents,
+			});
+
+			// since the CODE newsletters API isn't up to date with PROD newsletters API the
+			// review page will not show the correct newsletters on CODE.
+			// so when running in cypress we set a cookie to return the decrypted consents to cypress
+			// so we can check we at least got to the correct code path
+			if (runningInCypress) {
+				res.cookie(
+					'cypress-consent-response',
+					JSON.stringify(registrationConsents.consents),
+				);
+			}
```

Comment:
this is one of the examples of the fun cypress mocked cookie things to enable certain things on the server which i mentioned this morning 😛